src/SFA.DAS.LearnerDataMismatches.Web/Pages/Start.cshtml.cs

```diff
@@ -22,13 +30,26 @@ public IActionResult OnPost()
             }
         }
 
-        private void ValidateModel()
+        private async Task ValidateModel()
         {
             var isUlnANumber = long.TryParse(Uln, out long uln);
             if(Uln == null || isUlnANumber == false || uln <= 0 || uln > 9999999999)
             {
                 ModelState.AddModelError(nameof(Uln), "Enter a valid ULN");
             }
+            else
+            {
+                var apprenticeship = await _context
```

Comment:
Performing the "not found" validation here will result in two `select from Apprenticeship` queries immediately after each other - one on the validation of this page and then one on the data retrieval of the learner page.

Also, if someone were to manually navigate to `/learner/123456` we don't have any visibility that 123456 is not found.

Might it be worth putting the "not found" logic in the learner page itself?

Reply:
> Performing the "not found" validation here will result in two select from Apprenticeship queries immediately after each other - one on the validation of this page and then one on the data retrieval of the learner page.
Very good point, however to be able to show the validation error on search page, we cannot avoid the double query. 

>Also, if someone were to manually navigate to /learner/123456 we don't have any visibility that 123456 is not found.
Also a valid point, I think we need to run full validation on the learner page as well or change the way we navigate to it. Either way that will be a new story I guess.

Reply:
Is there a way the learner page can redirect the validation failure back to the seach page?  Some kind of redirect that pre-populates the `ModelState`.

I would be OK with not having the extra validation on the search page, and simply show a "Not Found" message on the learner page.  That's getting more into the UX designer's territory though.

Reply:
As agreed, I have changed the validation code to read the first record instead of all. I am still fetching the whole object as it makes it deterministic if it comes back null which on the other hand if we select any single field it gets difficult to determine if it comes back with a default value when using FirstOrDefault. 

A follow up story will be required for the Learner page for the scenario where a user directly navigates to it with invalid ULN.